test/client/index.test.js

```diff
@@ -1,73 +1,220 @@
+          .then(msg => {
+            assert.ok(typeof msg.id !== 'undefined');
+            done();
+          });
+      }).catch(done);
   });
-  
-  after(done => {
-    socket.once('disconnect', () => server.close(done));
-    socket.disconnect();
+
+  it('local authentication with wrong credentials fails', done => {
+    app.authenticate({
+        type: 'local',
+        email,
+        password: 'this is wrong'
+      })
+      .then(() => done(new Error('Should never get here')))
+      .catch(error => {
+        assert.equal(error.name, 'NotAuthenticated');
+        assert.equal(error.code, 401);
+        done();
+      });
   });
-  
-  it('adds .authenticate, .user, .token and .logout', () => {
-    const app = feathers().configure(authentication());
-    
-    assert.equal(typeof app.authenticate, 'function');
-    assert.equal(typeof app.user, 'function');
-    assert.equal(typeof app.token, 'function');
-    assert.equal(typeof app.logout, 'function');
+
+  it('authentication with no options and no stored token fails', done => {
+    app.authenticate()
+      .then(() => done(new Error('Should never get here')))
+      .catch(error => {
+        assert.equal(error.message, 'Could not find stored JWT and no authentication type was given');
+        done();
+      });
+  });
+
+  it('uses localStorage compatible stores', done => {
+    const oldStorage = app.get('storage');
+    app.set('storage', localstorage);
+
+    app.authenticate({
+        type: 'local',
+        email, password
+      }).then(response => {
+        assert.equal(response.token, localstorage.getItem('feathers-jwt'));
+        app.set('storage', oldStorage);
+        done();
+      });
   });
-  
-  it('authenticates via Socket.io', done => {
-    const app = feathers()
-      .configure(socketio(socket))
-      .configure(hooks())
-      .configure(authentication())
-      .use('/storage', memory());
-    
-    socket.on('connect', () => {
-      app.authenticate({
+
+  it('token is stored and re-authentication with stored token works', done => {
+    app.authenticate({
         type: 'local',
         email, password
       }).then(response => {
         assert.ok(response.token);
         assert.ok(response.data);
-        
-        app.user()
-          .then(user => {
-            assert.deepEqual(user, response.data);
-            
-            return app.token();
-          })
-          .then(token => assert.deepEqual(token, response.token))
-          .then(done);
+
+        return app.token().then(token => {
+          assert.deepEqual(response.token, token);
+        }).then(() => {
+          return app.authenticate().then(response => {
+            assert.deepEqual(app.get('token'), response.token);
+            assert.deepEqual(app.get('user'), response.data);
+          });
+        }).then(done);
       }).catch(done);
+  });
+
+  it.skip('.logout works, lets you not access to protected service', done => {
```

Comment:
Ah. Yeah, the problem was that this test fails with sockets because `app.logout` really has to disconnect the socket, too. I'll give it a shot with `.end()`.